networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/gossip/LibP2PGossipNetwork.java

```diff
@@ -13,39 +13,70 @@
 import io.libp2p.core.pubsub.PubsubPublisherApi;
 import io.libp2p.core.pubsub.PubsubSubscription;
 import io.libp2p.core.pubsub.Topic;
+import io.libp2p.core.pubsub.ValidationResult;
+import io.libp2p.pubsub.FastIdSeenCache;
+import io.libp2p.pubsub.PubsubRouterMessageValidator;
+import io.libp2p.pubsub.SeenCache;
+import io.libp2p.pubsub.TTLSeenCache;
 import io.libp2p.pubsub.gossip.Gossip;
+import io.libp2p.pubsub.gossip.GossipParams;
+import io.libp2p.pubsub.gossip.GossipRouter;
 import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandler;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import kotlin.jvm.functions.Function0;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.networking.p2p.gossip.GossipNetwork;
+import tech.pegasys.teku.networking.p2p.gossip.PreparedGossipMessage;
+import tech.pegasys.teku.networking.p2p.gossip.PreparedGossipMessageFactory;
 import tech.pegasys.teku.networking.p2p.gossip.TopicChannel;
 import tech.pegasys.teku.networking.p2p.gossip.TopicHandler;
 import tech.pegasys.teku.networking.p2p.libp2p.LibP2PNodeId;
+import tech.pegasys.teku.networking.p2p.network.GossipConfig;
 import tech.pegasys.teku.networking.p2p.peer.NodeId;
 
-public class LibP2PGossipNetwork implements tech.pegasys.teku.networking.p2p.gossip.GossipNetwork {
+public class LibP2PGossipNetwork implements GossipNetwork {
+
   private static final Logger LOG = LogManager.getLogger();
+  private static final PubsubRouterMessageValidator STRICT_FIELDS_VALIDATOR =
+      new GossipWireValidator();
+  private static final Function0<Long> NULL_SEQNO_GENERATOR = () -> null;
 
   private final MetricsSystem metricsSystem;
   private final Gossip gossip;
   private final PubsubPublisherApi publisher;
+  private final Map<String, TopicHandler> topicHandlerMap = new ConcurrentHashMap<>();
+  private final PreparedGossipMessageFactory defaultMessageFactory;
 
   public LibP2PGossipNetwork(
-      final MetricsSystem metricsSystem, final Gossip gossip, final PubsubPublisherApi publisher) {
+      MetricsSystem metricsSystem,
+      GossipConfig gossipConfig,
+      PreparedGossipMessageFactory defaultMessageFactory,
+      boolean logWireGossip) {
     this.metricsSystem = metricsSystem;
-    this.gossip = gossip;
-    this.publisher = publisher;
+    this.defaultMessageFactory = defaultMessageFactory;
+    this.gossip = createGossip(gossipConfig, logWireGossip);
+    this.publisher = gossip.createPublisher(null, NULL_SEQNO_GENERATOR);
```

Comment:
Good idea to move this init out of `LibP2pNetwork` but might be better to introduce a static `create` method rather than doing all the work in the constructor.

Reply:
Yep, totally makes sense with the final PR version (this was done for the previous PR incarnation)

Reply:
Please check this change: fa656bb. Is that what you have in mind?